packages/maskbook/src/plugins/Wallet/hooks/useCollectibles.ts

```diff
@@ -30,3 +32,19 @@ export function useCollectiblesFromNetwork(address: string, provider: Collectibl
         return WalletRPC.getAssetsListNFT(address.toLowerCase(), provider)
     }, [address, provider])
 }
+
+export function useCollectibles(address: string, provider: CollectibleProvider, page: number) {
+    const values = useRef<Collectible[]>([])
+    return useAsyncRetry(async () => {
+        if (page === 1) values.current = []
+
+        if (!address) return []
+        const result = await WalletRPC.getAssetsListNFT(address.toLowerCase(), provider, page)
+        const erc721Tokens = await WalletRPC.getERC721TokensPaged(page, 50)
+
+        values.current.push(
+            ...uniqBy([...result, ...erc721Tokens.map((x) => ERC721TokenRecordToCollectible(x))], 'token_id'),
```

Comment:
Both `address` and `token_id` should be used to define an unique ERC721 token. Different tokens from different contracts may got same `token_id`.

Reply:
```suggestion
            ...uniqBy([...result, ...erc721Tokens.map(ERC721TokenRecordToCollectible)], 'token_id'),
```